crop-planner/src/Table.tsx

```diff
@@ -1,71 +1,140 @@
-import { PureComponent } from 'react'
-import { sortBy } from 'lodash'
+import { PureComponent } from "react";
+import { sortBy } from "lodash";
 
-import CropSelect from './CropSelect'
-import { Crop, Field, SeasonalCrop } from './types'
-import { fetchCrops, fetchFields } from './api'
-import buildNewFieldsState from './buildNewFieldsState'
+import CropSelect from "./CropSelect";
+import { Crop, Field, SeasonalCrop, HumusBalance } from "./types";
+import { fetchCrops, fetchFields, fetchHumusBalance } from "./api";
+import buildNewFieldsState from "./buildNewFieldsState";
 
-type Props = {}
+type Props = {};
 
 type State = {
-  allCrops: Array<Crop>,
-  fields: Array<Field>
-}
+  allCrops: Array<Crop>;
+  fields: Array<Field>;
+  humusBalances: { [key: number]: number };
+};
 
 export default class Table extends PureComponent<Props, State> {
   constructor(props: Props) {
-    super(props)
+    super(props);
 
     this.state = {
       allCrops: [],
       fields: [],
-    }
+      humusBalances: {},
+    };
   }
 
-  componentDidMount = async () =>
+  updateInitialHumusBalance(field: Field, balance: number): void {
+    const humusBalances = {
+      ...this.state.humusBalances,
+    };
+    humusBalances[field.id] = balance;
     this.setState({
-      fields: await fetchFields(),
-      allCrops: await fetchCrops(),
-    })
-
-  render = () =>
-    <div className="table">
-      <div className="table__row table__row--header">
-        <div className="table__cell">Field name</div>
-        <div className="table__cell table__cell--right">Field area (ha)</div>
-        <div className="table__cell table__cell--center">2020 crop</div>
-        <div className="table__cell table__cell--center">2021 crop</div>
-        <div className="table__cell table__cell--center">2022 crop</div>
-        <div className="table__cell table__cell--center">2023 crop</div>
-        <div className="table__cell table__cell--center">2024 crop</div>
-        <div className="table__cell table__cell--right">Humus balance</div>
-      </div>
+      humusBalances: humusBalances,
```

Comment:
All this could be done with a single expression:
```
    this.setState((prevState) => ({
      humusBalances: {
        ...prevState.humusBalances,
       [field.id]: balance,
      }
    }))
```